File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_006)_Left_Rotate_Array_by_d_(Time_dN).cpp
#include <iostream>
#include <cmath>
using namespace std;

void lRotateOne(int arr[], int n)
{
	int temp = arr[0];
	for(int i = 1; i < n; i++)
	{
		arr[i - 1] = arr[i];
	}
	arr[n - 1] = temp;
}
void leftRotate(int arr[], int d, int n)
{
	for(int i = 0; i < d; i++)
	{
		lRotateOne(arr, n);
	}
}
int main()
{
       int arr[] = {1, 2, 3, 4, 5}, n = 5, d = 2;
       cout<<"Before Left Rotation"<<endl;
       for(int i = 0; i < n; i++)
       {
       		cout<<arr[i]<<" ";
       }
       cout<<endl;
       leftRotate(arr, d, n);
       cout<<"After '2' Left Rotation"<<endl;
       for(int i = 0; i < n; i++)
       {
       		cout<<arr[i]<<" ";
       }
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_006)_Left_Rotate_Array_by_d_(Time_dN).dart
void leftRotateOne(arr, int n) {
  var temp = arr[0];
  for (int i = 1; i < n; i++) {
    arr[i - 1] = arr[i];
  }
  arr[n - 1] = temp;
}

void leftRotate(arr, int d, int n) {
  for (int i = 0; i < d; i++) {
    leftRotateOne(arr, n);
  }
}

void main(List<String> args) {
  var arr = [1, 2, 3, 4, 5];
  print('Before Left Rotate: ${arr}');
  leftRotate(arr, 2, arr.length);
  print('After Left Rotate: ${arr}');
}
